File: [프로그래머스] 연습문제_콜라츠 추측.py
def solution(num):
    cnt = 0
    while True:
        if cnt == 500:
            cnt = -1
            break
        elif num == 1:
            break
        elif num%2 == 0:
            num = num / 2
            cnt += 1
        elif num%2 != 0:   
            num = num * 3 + 1
            cnt += 1
    return cnt

File: [프로그래머스] 연습문제_콜라츠 추측.js
function solution(num) {
    let answer = 0;
    if(num === 1) return 0;
    while(answer <= 500){
        if(answer === 500) return -1;
        num % 2 ? num = num * 3 + 1 : num /= 2
        answer++;
        if(num === 1) return answer;
    }
    return answer;
}